reactive/rerunner.go

```diff
@@ -225,24 +174,46 @@ func Cache(ctx context.Context, key interface{}, f ComputeFunc) (interface{}, er
 	}
 
 	cache := ctx.Value(cacheKey{}).(*cache)
-	computation := ctx.Value(computationKey{}).(*computation)
+	parent := ctx.Value(computationKey{}).(*computation)
 
-	cache.locker.Lock(key)
-	defer cache.locker.Unlock(key)
-
-	if child := cache.get(key); child != nil {
-		child.node.addOut(&computation.node)
-		return child.value, nil
+	cache.mu.Lock()
```

Comment:
do we want to use a sync.Map or RWMutex here?  (Might be less contention when reading entries).